packages/editor/src/components/rich-text/index.native.js

```diff
@@ -274,8 +274,8 @@ export class RichText extends Component {
 
 		// If the component is changed React side (undo/redo/merging/splitting/custom text actions)
 		// we need to make sure the native is updated as well
-		if ( nextProps.value &&
-			this.lastContent &&
+		if ( ( typeof nextProps.value !== 'undefined' ) &&
```

Comment:
I think this is normally done with `nextProps.value !== undefined` instead of `typeof` but I'm not 100% sure